File: src/kyu_6/make_a_window/python/make_a_window.py
def make_a_window(n):
    top     = '-' * (2*n+3)
    middle  = f"|{ '-'*n }+{ '-'*n }|"
    glasses = [f"|{ '.'*n }|{ '.'*n }|"] * n
    return '\n'.join([top, *glasses, middle, *glasses, top])

File: src/kyu_6/make_a_window/csharp/make_a_window.cs
using System;
using System.Linq;

static class Kata
{
    public static string MakeAWindow(int n)
    {
      var win = "|" + new string('.', n) + "|" + new string('.', n) + "|" + "\n";
      var middle = "|" + new string('-', n) + "+" + new string('-', n) + "|" + "\n";
      var topBottom = new string('-', win.Length-1) + "\n";
      return (topBottom + string.Concat(Enumerable.Range(0, n).Select(x => win)) + middle + string.Concat(Enumerable.Range(0, n).Select(x => win)) + topBottom).Trim('\n');
    }
}